bigtable/tests/mutation_integration_test.cc

```diff
@@ -0,0 +1,508 @@
+  auto micro_secs =
+      ((std::chrono::system_clock::now().time_since_epoch().count() + 50000) /
+       1000000) *
+      1000;
+  // Add one millisecond as processing to match the server timestamp
+  micro_secs += 1000;
+  std::vector<bigtable::Cell> expected_cells{
+      {row_key, column_family1, "column_id1", micro_secs, "v-c-0-0", {}},
+      {row_key, column_family1, "column_id2", micro_secs, "v-c-0-1", {}},
+      {row_key, column_family1, "column_id3", micro_secs, "v-c-0-2", {}},
+      {row_key, column_family2, "column_id2", micro_secs, "v-c0-0-0", {}},
+      {row_key, column_family2, "column_id3", micro_secs, "v-c1-0-1", {}},
+      {row_key, column_family3, "column_id1", micro_secs, "v-c1-0-2", {}},
+  };
+
+  CreateCellsWithoutTimestamp(*table, created_cells);
+
+  auto actual_cells = ReadRows(*table, bigtable::Filter::PassAllFilter());
+
+  DeleteTable(table_name);
+
+  CheckEqualUnorderedWOTimestamp(expected_cells, actual_cells);
+}
+
+/**
+ * Check if list of table names are matching with the
+ * expected tablename list
+ */
+TEST_F(MutationIntegrationTest, DeleteFromColumnForTimestampTest) {
+  std::string const table_name = "table4delcol";
+
+  auto table = CreateTable(table_name, table_config);
+
+  // Create a vector of cell which will be inserted into bigtable
+  std::string const row_key = "DelColKey";
+  std::vector<bigtable::Cell> created_cells{
+      {row_key, column_family1, "column_id1", 0, "v-c-0-0", {}},
+      {row_key, column_family1, "column_id2", 1000, "v-c-0-1", {}},
+      {row_key, column_family1, "column_id3", 2000, "v-c-0-2", {}},
+      {row_key, column_family2, "column_id2", 2000, "v-c0-0-0", {}},
+      {row_key, column_family2, "column_id2", 1000, "v-c0-0-1", {}},
+      {row_key, column_family2, "column_id2", 3000, "v-c0-0-2", {}},
+      {row_key, column_family2, "column_id2", 4000, "v-c0-0-3", {}},
+      {row_key, column_family2, "column_id3", 1000, "v-c1-0-1", {}},
+      {row_key, column_family2, "column_id2", 2000, "v-c1-0-2", {}},
+      {row_key, column_family3, "column_id1", 2000, "v-c1-0-2", {}},
+  };
+
+  // timestamp_end is not inclusive.
+  std::int64_t timestamp_begin = 2000;
+  std::int64_t timestamp_end = 4000;
+  std::vector<bigtable::Cell> expected_cells{
+      {row_key, column_family1, "column_id1", 0, "v-c-0-0", {}},
+      {row_key, column_family1, "column_id2", 1000, "v-c-0-1", {}},
+      {row_key, column_family1, "column_id3", 2000, "v-c-0-2", {}},
+      {row_key, column_family2, "column_id2", 1000, "v-c0-0-1", {}},
+      {row_key, column_family2, "column_id2", 4000, "v-c0-0-3", {}},
+      {row_key, column_family2, "column_id3", 1000, "v-c1-0-1", {}},
+      {row_key, column_family3, "column_id1", 2000, "v-c1-0-2", {}},
+  };
+
+  // Create records
+  CreateCells(*table, created_cells);
+
+  // Delete the columns with column identifier as created_cells.at(3) (i.e.
+  // column_id2)
+  DeleteFromColumn(*table, created_cells.at(3), timestamp_begin, timestamp_end);
```

Comment:
This is a bit obscure, why not just `DeleteFromColumn(*table, "column_id3", timestamp_begin, timestamp_end);` ?